docs/source/daml/reference/choices.rst

```diff
@@ -72,10 +72,59 @@ Controllers
 
   The conjunction of **all** the parties are required to authorize when this choice is exercised.
 
-.. _daml-ref-anytime:
+.. _daml-ref-consumability:
+
+Consumability
+=============
+
+If no qualifier is present, choices are *consuming*: the contract is archived before the evaluation of the choice body.
+
+Note that the contracts created in the choice body are divulged to the signatories and controllers of the contracts as well as to the observers of the contract on which the choice was exercised. This behavior differs from that of *preconsuming* and *postconsuming* choices.
+
+Preconsuming choices
+********************
+
+.. literalinclude:: ../code-snippets/Reference.daml
+   :language: daml
+   :start-after: -- start choice-first preconsuming snippet
+   :end-before: -- end choice-first preconsuming snippet
+   :caption: Option 1 for specifying choices: choice name first
+
+.. literalinclude:: ../code-snippets/Reference.daml
+   :language: daml
+   :start-after: -- start controller-first preconsuming snippet
+   :end-before: -- end controller-first preconsuming snippet
+   :caption: Option 2 for specifying choices: controller first
+
+- ``preconsuming`` keyword. Optional.
+- Makes a choice pre-consuming: the contract is archived before the body of the exercise is executed.
+- The archival behavior is analogous to the *consuming* default behavior.
+- Unlike what happens the in *consuming* default behavior, though, contracts created in the choice body are divulged only to their respective signatories and controllers and not to the observers of the contract on which the choice was exercised.
+- Can be thought as a non-consuming choice that implicitly archives the contract before anything else happens
+
+Postconsuming choices
+*********************
+
+.. literalinclude:: ../code-snippets/Reference.daml
+   :language: daml
+   :start-after: -- start choice-first postconsuming snippet
+   :end-before: -- end choice-first postconsuming snippet
+   :caption: Option 1 for specifying choices: choice name first
+
+.. literalinclude:: ../code-snippets/Reference.daml
+   :language: daml
+   :start-after: -- start controller-first postconsuming snippet
+   :end-before: -- end controller-first postconsuming snippet
+   :caption: Option 2 for specifying choices: controller first
+
+- ``postconsuming`` keyword. Optional.
+- Makes a choice post-consuming: the contract is archived after the body of the exercise is executed.
+- The contract can still be used in the body of the exercise.
+- Contracts created in the choice body are divulged only to their respective signatories and controllers and not to the observers of the contract on which the choice was exercised.
```

Comment:
And here

Reply:
Solved in https://github.com/digital-asset/daml/pull/4134/commits/11f52f472e3cb698771d149a1e0d4ac0a464f02a